File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/nullable/KNullableBooleanPropertyName.java
package com.nagopy.android.kotlinames.property.nullable;

import com.nagopy.android.kotlinames.KNullablePropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KNullableBooleanPropertyName implements KNullablePropertyName<Boolean>, KSortablePropertyName<Boolean> {

    private final String name;

    public KNullableBooleanPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/nullable/KNullableBooleanPropertyName.kt
package com.nagopy.android.kotlinames.property.nullable

import com.nagopy.android.kotlinames.KNullablePropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KNullableBooleanPropertyName(val name: String) : KNullablePropertyName<Boolean>, KSortablePropertyName<Boolean> {

    override fun name(): String = name

}
